Mods/Editor/Src/Components/EntityTree.cpp

```diff
@@ -377,15 +379,36 @@ void Editor::DrawEntityTree() {
 	}
 }
 
-void Editor::OnSelectEntity(ZEntityRef p_Entity, std::optional<std::string> p_ClientId) {
+void Editor::OnSelectEntity(ZEntityRef p_Entity, std::optional<std::string> p_ClientId)
+{
 	const bool s_DifferentEntity = m_SelectedEntity != p_Entity;
 
 	m_SelectedEntity = p_Entity;
 	m_ShouldScrollToEntity = p_Entity.GetEntity() != nullptr;
 
-	if (s_DifferentEntity) {
+	if (s_DifferentEntity)
+	{
 		m_Server.OnEntitySelected(p_Entity, std::move(p_ClientId));
 	}
+
+	if (!m_SelectionForEntityCreated)
+	{
+		s_SelectionForFreeCameraEditorStyleEntity2 = std::unique_ptr<unsigned char[]>(new unsigned char[sizeof(ZSelectionForFreeCameraEditorStyleEntity)]);
+		s_SelectionForFreeCameraEditorStyleEntity = reinterpret_cast<ZSelectionForFreeCameraEditorStyleEntity*>(s_SelectionForFreeCameraEditorStyleEntity2.get());
+		entityRef.m_pInterfaceRef = s_SelectionForFreeCameraEditorStyleEntity;
```

Comment:
I asked Pavle since they were the ones to help with this. They are saying that we can't use a constructor because we can't create ZEntityType since it requires using a custom CPPT/CBLU for the `ZSelectionForFreeCameraEditorStyleEntity` class which would require using a SMF mod.

Reply:
Do you have any ideas on what I can do?

Reply:
You can call the constructor with a pointer to a null ZEntityType.